BO4ETestProject/TestUserProperties.cs

```diff
@@ -119,5 +120,63 @@ public void TestFlags()
             melo.UserProperties["foo"] = null;
             Assert.IsFalse(melo.HasFlagSet("foo"));
         }
+
+        /// <summary>
+        /// Tests to set and get a UserProperty
+        /// </summary>
+        [TestMethod]
+        public void TestSetterGetter()
+        {
+            var melo = new Messlokation
+            {
+                MesslokationsId = "DE0123456789012345678901234567890",
+                Sparte = BO4E.ENUM.Sparte.STROM
+            };
+            Assert.IsNull(melo.UserProperties);
+
+            var value1 = "BarFoo";
+            var value2 = "BigFoot";
+            melo.SetUserProperty("foo", new Bar() { FooBar = value1 });
+            Assert.IsNotNull(melo.UserProperties);
+            Assert.IsTrue(melo.UserProperties.TryGetValue("foo", out var upValue));
+            Assert.AreEqual(value1, (upValue as Bar)?.FooBar);
+
+            // Update the value
+            melo.SetUserProperty("foo", new Bar() { FooBar = value2 });
+            Assert.IsTrue(melo.UserProperties.TryGetValue("foo", out upValue));
+            Assert.AreEqual(value2, (upValue as Bar)?.FooBar);
+        }
+
+        /// <summary>
+        /// Tests to remove UserProperty
+        /// </summary>
+        [TestMethod]
+        public void TestDeletion()
+        {
+            var melo = new Messlokation
+            {
+                MesslokationsId = "DE0123456789012345678901234567890",
+                Sparte = BO4E.ENUM.Sparte.STROM
+            };
+            Assert.IsNull(melo.UserProperties);
+
+            var value1 = "BarFoo";
+            // Add a value
+            melo.SetUserProperty("foo", new Bar() { FooBar = value1 });
+            Assert.IsNotNull(melo.UserProperties);
+            Assert.IsTrue(melo.UserProperties.TryGetValue("foo", out var upValue));
+            Assert.AreEqual(value1, (upValue as Bar)?.FooBar);
+            // remove it again
+            melo.RemoveUserProperty("foo");
+            Assert.IsFalse(melo.UserProperties.TryGetValue("foo", out upValue));

```

Comment:
```suggestion
            Assert.IsFalse(melo.UserProperties.TryGetValue("foo", out upValue));
            Assert.IsNotNull(melo.UserProperties);
```
dann ist auch klar, dass das Ding als leeres dict stehen bleibt.